msktutil.cpp

```diff
@@ -668,7 +668,8 @@ int execute(msktutil_exec *exec, msktutil_flags *flags)
         remove_keytab_entries(flags, exec->remove_principals);
 
         /* update keytab */
-        if (!flags->dont_change_password) {
+        if (!flags->dont_change_password ||
+            (flags->dont_change_password && exec->mode == MODE_CREATE)) {
```

Comment:
There's a gotcha with these checks, unfortunately. As stated in #91, `create` is supposed to be identical to `update --service ...`. While this may be true in terms of functionality, the sanity check here will trigger for `create`, but not if you (or more likely: your automation scripts) rely on `update --service` doing the same thing.